lib/vdsm/storage/lvm.py

```diff
@@ -577,6 +577,73 @@ def _getVGDevs(self, vgNames):
                 devices = tuple(devices)
         return devices
 
+    def _updatevgs(self, vgs_output, vg_names):
```

Comment:
Maybe to make it more clear that this must be called while holding the lock, call the helpers:

    _update_vgs_locked()
    _update_stale_vgs_locked()